pkg/controller/cell.go

```diff
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package controller
+
+import (
+	"github.com/spf13/pflag"
+
+	"github.com/cilium/cilium/pkg/hive/cell"
+	"github.com/cilium/cilium/pkg/metrics"
+	"github.com/cilium/cilium/pkg/metrics/metric"
+)
+
+const (
+	controllerGroupMetrics = "controller-group-metrics"
+
+	// labelControllerGroupName is the label used
+	// to identify controller-specific metrics
+	labelControllerGroupName = "group_name"
+)
+
+var (
+	// GroupMetricEnabled is populated with the set of ControllerGroups for which metrics are enabled
+	groupMetricEnabled = map[string]bool{}
+
+	// GroupRuns is a Prometheus-compatible metric for Controller
+	// runs, labeled by completion status and Group name
+	GroupRuns metric.Vec[metric.Counter]
+)
+
+var Cell = cell.Module(
+	"controller",
+	"Controllers and Controller Lifecycle management",
+	cell.Config(Config{}),
+	cell.Metric(NewMetrics),
+	cell.Invoke(Init),
+)
+
+type Metrics struct {
+	ControllerGroupRuns metric.Vec[metric.Counter]
+}
+
+func NewMetrics() Metrics {
```

Comment:
We should add the new metrics in the metrics reference in the docs: https://docs.cilium.io/en/v1.14/observability/metrics/#metrics-reference